tests/ghost_test.go

```diff
@@ -0,0 +1,211 @@
+	migrationContext.SetDMLBatchSize(dmlBatchSize)
+	migrationContext.SetMaxLagMillisecondsThrottleThreshold(maxLagMillisecondsThrottleThreshold)
+	migrationContext.SetDefaultNumRetries(defaultNumRetries)
+	migrationContext.ApplyCredentials()
+	if err := migrationContext.SetCutOverLockTimeoutSeconds(cutoverLockTimoutSeconds); err != nil {
+		return nil, err
+	}
+	if err := migrationContext.SetExponentialBackoffMaxInterval(exponentialBackoffMaxInterval); err != nil {
+		return nil, err
+	}
+	return migrationContext, nil
+}
+
+func TestGhostSimpleNoop(t *testing.T) {
+	const (
+		localhost = "127.0.0.1"
+		port      = 13306
+		user      = "root"
+		database  = "gh_ost_test_db"
+		table     = "tbl"
+	)
+	err := func() error {
+		basedir := t.TempDir()
+		datadir := filepath.Join(basedir, "data")
+		if err := os.Mkdir(datadir, 0755); err != nil {
+			return err
+		}
+		instance, err := mysql.Install(basedir, datadir, "root")
+		if err != nil {
+			return fmt.Errorf("failed to start MySQL: %v", err)
+		}
+		if err := instance.Start(port, os.Stdout, os.Stderr, 60); err != nil {
+			return fmt.Errorf("failed to start MySQL: %v", err)
+		}
+
+		defer func() {
+			_ = instance.Stop(os.Stdout, os.Stderr)
+		}()
+
+		db, err := sql.Open("mysql", fmt.Sprintf("root@tcp(localhost:%d)/mysql", port))
+		if err != nil {
+			return fmt.Errorf("failed to open MySQL: %v", err)
+		}
+		defer db.Close()
+
+		if _, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", database)); err != nil {
+			return err
+		}
+
+		if _, err = db.Exec(fmt.Sprintf("USE %s", database)); err != nil {
+			return err
+		}
+
+		if _, err = db.Exec(fmt.Sprintf("CREATE TABLE %s (id int primary key, data int)", table)); err != nil {
+			return err
+		}
+
+		tx, err := db.Begin()
+		if err != nil {
+			return err
+		}
+		defer tx.Rollback()
+
+		const n = 100
+
+		for i := 1; i <= n; i++ {
+			_, err := tx.Exec(fmt.Sprintf("INSERT INTO %s VALUES (%v, %v)", table, i, i))
+			if err != nil {
+				return err
+			}
+		}
+
+		if err := tx.Commit(); err != nil {
+			return err
+		}
+
+		migrationContext, err := newMigrationContext(config{
+			host:           localhost,
+			port:           port,
+			user:           user,
+			database:       database,
+			table:          table,
+			alterStatement: "ALTER TABLE tbl ADD name VARCHAR(64)",
+			noop:           true,
+		})
+
+		if err != nil {
+			return err
+		}
+
+		migrator := logic.NewMigrator(migrationContext)
+		if err := migrator.Migrate(); err != nil {
+			return err
+		}
+
+		rows, err := db.Query(`SELECT * FROM tbl`)
+		if err != nil {
+			return err
+		}
+		defer rows.Close()
+
+		var (
+			id   int
+			data int
+		)
+
+		rowCount := 0
+
+		for rows.Next() {
+			if err := rows.Scan(&id, &data); err != nil {
+				return err
+			}
+			if id != data {
+				return fmt.Errorf("data mismatch, id != data, id: %v, data: %v", id, data)
+			}
+			rowCount++
+		}
+
+		if rowCount != n {
+			return fmt.Errorf("expect row count: %v, get %v", n, rowCount)
```

Comment:
Try to use the same format for these

e.g. say "row count mismatch, got: xxx, want: yyy" instead of "expect xxx xxx: